<a href="https://colab.research.google.com/github/Zelenka777/project_01/blob/main/1.4.ipynb" target="_parent"><img src="https://colab.research.google.com/assets/colab-badge.svg" alt="Open In Colab"/></a>

In [1]:
titles = {
    'Кроссовки тип 3 (Adidas)': '100000110',
    'Мячик тип 2 (Adidas)': '100000146',
    'Кепка тип 1 (Adidas)': '100000149',
    'Ремень тип 2 (Nike)': '100000194',
    'Футболка тип 1 (Adidas)': '100000224',
    'Шапка тип 5 (Puma)': '100000280',
  }

store = { 
    '100000110': [
        {'quantity': 31, 'price': 1637}
        ], 
    '100000146': [ 
        {'quantity': 4, 'price': 45}, 
        {'quantity': 10, 'price': 48}
         ], 
    '100000149': [ 
        {'quantity': 28, 'price': 279}, 
        {'quantity': 32, 'price': 291}
        ], 
    '100000194': [
        {'quantity': 8, 'price': 220}, 
        {'quantity': 1, 'price': 170}
        ], 
    '100000224': [
        {'quantity': 61, 'price': 438}, 
        {'quantity': 23, 'price': 302}, 
        {'quantity': 50, 'price': 412}
        ], 
    '100000280': [
        {'quantity': 26, 'price': 175}, 
        ] 
    }


for title in titles:
    quantity = 0
    title_id = titles[title]
    sum_price = 0
    for item in store[title_id]:
        quantity += item['quantity']
        sum_price += item['quantity'] * item['price']
    print(f"{title} - {quantity} шт, стоимость {sum_price} руб")



Кроссовки тип 3 (Adidas) - 31 шт, стоимость 50747 руб
Мячик тип 2 (Adidas) - 14 шт, стоимость 660 руб
Кепка тип 1 (Adidas) - 60 шт, стоимость 17124 руб
Ремень тип 2 (Nike) - 9 шт, стоимость 1930 руб
Футболка тип 1 (Adidas) - 134 шт, стоимость 54264 руб
Шапка тип 5 (Puma) - 26 шт, стоимость 4550 руб


# Задача 1.4.

# Есть словарь кодов товаров titles

titles = {
    'Кроссовки тип 3 (Adidas)': '100000110',
    'Мячик тип 2 (Adidas)': '100000146',
    'Кепка тип 1 (Adidas)': '100000149',
    'Ремень тип 2 (Nike)': '100000194',
    'Футболка тип 1 (Adidas)': '100000224',
    'Шапка тип 5 (Puma)': '100000280',
}

# Товары находятся на складе и сохранены в виде словаря списков словарей,
# которые отражают количество товаров в магазине по каждому коду.

store = {
    '100000110': [{'quantity': 31, 'price': 1637}],
    '100000146': [ {'quantity': 4, 'price': 45}, {'quantity': 10, 'price': 48}],
    '100000149': [ {'quantity': 28, 'price': 279}, {'quantity': 32, 'price': 291}],
    '100000194': [{'quantity': 8, 'price': 220}, {'quantity': 1, 'price': 170}],
    '100000224': [{'quantity': 61, 'price': 438}, {'quantity': 23, 'price': 302},  {'quantity': 50, 'price': 412}],
    '100000280': [{'quantity': 26, 'price': 175}, ]
}

# Рассчитайте на какую сумму лежит каждого товара на складе.
#
# Вывести суммарную стоимость каждого товара в магазине в формате:
# "<товар> - <кол-во> шт, стоимость <общая стоимость> руб"

# Пример: "Кроссовки тип 3 (Adidas) - 31 шт, стоимость 50747 руб"